File: app_pojavlauncher/src/main/java/net/kdt/pojavlaunch/customcontrols/gamepad/DefaultDataProvider.java
package net.kdt.pojavlaunch.customcontrols.gamepad;


import static net.kdt.pojavlaunch.platform.Platform.PLATFORM;

import net.kdt.pojavlaunch.platform.input.PlatformGrabListener;
import net.kdt.pojavlaunch.platform.Platform;


public class DefaultDataProvider implements GamepadDataProvider {
    public static final DefaultDataProvider INSTANCE = new DefaultDataProvider();

    // Cannot instantiate this class publicly
    private DefaultDataProvider() {}

    @Override
    public GamepadMap getGameMap() {
        return GamepadMapStore.getGameMap();
    }


    @Override
    public GamepadMap getMenuMap() {
        return GamepadMapStore.getMenuMap();
    }

    @Override
    public boolean isGrabbing() {
        // Avoid going through the JNI each time.
        return Platform.isGrabbing();
    }

    @Override
    public void attachGrabListener(PlatformGrabListener grabListener) {
        Platform.addGrabListener(grabListener);
    }
}

File: app_pojavlauncher/src/main/java/net/kdt/pojavlaunch/customcontrols/gamepad/DefaultDataProvider.kt
package net.kdt.pojavlaunch.customcontrols.gamepad

import git.artdeell.dnbootstrap.glfw.GLFW
import git.artdeell.dnbootstrap.glfw.GrabListener

object DefaultDataProvider : GamepadDataProvider {
    override fun getGameMap(): GamepadMap = GamepadMapStore.getGameMap()

    override fun getMenuMap(): GamepadMap = GamepadMapStore.getMenuMap()

    override fun isGrabbing(): Boolean = GLFW.isGrabbing()

    override fun attachGrabListener(grabListener: GrabListener) {
        GLFW.addGrabListener(grabListener)
    }
}
